src/app/services/items.service.ts

```diff
@@ -0,0 +1,94 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable, find, map, of, switchMap, tap } from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { Item } from '../models/item';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ItemsService {
+  items$: Observable<Item[]> = this.http.get<Item[]>(`${environment.apiUrl}groceries`)
+  private itemCountsSubject = new BehaviorSubject(new Map<string, number>())
```

Comment:
If you had created a separate array to maintain the items in the cart, you may have been able to easily show the item data (such as the images)